commands/azureCommands/pull-from-azure.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Registry } from "azure-arm-containerregistry/lib/models";
+import { exec } from 'child_process';
+import { AzureImageTagNode, AzureRepositoryNode } from '../../explorer/models/azureRegistryNodes';
+import { ext } from '../../extensionVariables';
+import * as acrTools from '../../utils/Azure/acrTools';
+import { AzureImage } from "../../utils/Azure/models/image";
+import { Repository } from "../../utils/Azure/models/repository";
+import * as quickPicks from '../utils/quick-pick-azure';
+
+/* Pulls an image from Azure. The context is the image node the user has right clicked on */
+export async function pullFromAzure(context?: AzureImageTagNode | AzureRepositoryNode): Promise<any> {
+    let registryName: string;
+    let registry: Registry;
+    let imageName: string;
+
+    if (context) { // Right Click
+        registryName = context.registry.loginServer;
+        registry = context.registry;
+
+        if (context.contextValue === "azureImageTagNode") { // Right Click on AzureImageNode
```

Comment:
if (context instanceof AzureImageTagNode) {
…
} else if (context instanceof AzureRepositoryNode) {
…
} else {
assert.fail(`Unexpected node type: ${context.contextValue}`);
}

Reply:
I can't print context.contextValue in the else statement. tslint says "Property does not exist on type 'never'." It does a flow check and won't hit this statement- in theory due to the parameters of the function. I did  "Unexpected node type: ${context}"